File: Python3/프로그래머스/1/42748. K번째수/K번째수.py
def solution(array, commands):
    #i번째 숫자부터 j번째 숫자까지 자르고 정렬 -> K번째 있는 수
    #i,j,k
    answer = []
    for i,j,k in commands:
        temp = array[i-1:j]
        temp.sort()
        answer.append(temp[k-1])
    
    return answer
        
    
    
    
    
    
    
    
    
    
    
    
    
    
#     answer = []
    
#     for command in commands:
#         [i,j,k] = command
#         temp_array = array[i-1:j]
#         temp_array.sort()
#         answer.append(temp_array[k-1])
    
#     return answer

File: C++/프로그래머스/1/42748. K번째수/K번째수.cpp
#include <string>
#include <vector>
#include <algorithm>

using namespace std;
//2번째부터 5번째까지 자르고, 정렬, k번째 수
vector<int> solution(vector<int> array, vector<vector<int>> commands) {
    vector<int> answer;

    for (const auto& command : commands) {
        vector<int> temp(array.begin() + command[0] -1, array.begin() + command[1]);
        sort(temp.begin(), temp.end());
        answer.push_back(temp[command[2]-1]);
    }
    
    return answer;
}